### DISCUSSION TITLE
HRRR vertical wind shear units?

### DISCUSSION THREAD

**Discussion by adair-kovac** [NOAA-GSL/GSL-Regional-Model-Forum#22] [Q&A]
A few months back someone was saying they thought the units for vucsh and vvcsh listed in the model output (1/s) are wrong.

```
135:Vertical u-component shear:s**-1 (instant):lambert:heightAboveGroundLayer:levels 0-1000 m:fcst time 0 hrs:from 202201310000
137:Vertical u-component shear:s**-1 (instant):lambert:heightAboveGroundLayer:levels 0-6000 m:fcst time 0 hrs:from 202201310000
136:Vertical v-component shear:s**-1 (instant):lambert:heightAboveGroundLayer:levels 0-1000 m:fcst time 0 hrs:from 202201310000
138:Vertical v-component shear:s**-1 (instant):lambert:heightAboveGroundLayer:levels 0-6000 m:fcst time 0 hrs:from 202201310000
```

I notice that in the tech memo [[pdf]](https://rapidrefresh.noaa.gov/Diag-vars-NOAA-TechMemo.pdf) they're plotted as knots but it doesn't say what unit they're stored as. Presumably it's m/s. Can the metadata in the model output be updated and/or the correct units documented?

**Comment by EricJames-NOAA** [comment]
Thanks for the question!  The GRIB2 labeling for wind shear is correct; vertical wind shear is defined as a wind speed difference (in m/s) divided by a difference in height (in m), so the fundamental units are s-1.  But for the graphics (as reference in the Tech Memo), we multiply by the depth of the respective shear layers to express in terms of kts.  

Here's some documentation on the momentum variable units in GRIB2:

https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-2.shtml

**Reply by adair-kovac** [reply]
Thanks, that document will be helpful!

However, I'm still not understanding how this squares with the data. For instance, looking at this message:

```
136:Vertical v-component shear:s**-1 (instant):lambert:heightAboveGroundLayer:levels 0-1000 m:fcst time 0 hrs:from 202201310000
```

And calculating 10th and 90th percentile:

```
np.percentile(vvcsh, [10, 90])

> array([-5.14175797,  3.54574203])
```

Since this is a 1000-m layer, we're ending up with windspeeds on the order of 1 km/s or even 10 km/s?

**Reply by EricJames-NOAA** [reply]
Thanks, yes, you make a good point.  We need to double check the encoding of these variables.  I'm wondering if the values are being passed in to the GRIB2 files as if the units are m/s.  Let me do some digging into the post-processor, and also look at some other NCEP models.
